src/System.Threading.Thread/tests/MTAMain/MTAMain.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+namespace MTAMain
+{
+    internal static class Program
+    {
+        [MTAThread]
+        static int Main(string[] args)
+        {
+            string mode = args[0];
+            int retValue = 1;
+            Thread curThread = Thread.CurrentThread;
+
+            if (mode == "GetApartmentState")
+            {
+                if (curThread.GetApartmentState() == ApartmentState.MTA)
+                {
+                    curThread.SetApartmentState(ApartmentState.MTA);
+                    retValue = 0;
+                }
+
+            }
+            else
+            {
+                try
+                {
+                    curThread.SetApartmentState(ApartmentState.STA);
+                }
+                catch (InvalidOperationException)
+                {
+                    retValue = 0;
+                }
+                catch (PlatformNotSupportedException) {}
+            }
```

Comment:
Nit, OK to leave it here, but defensive programming would check for expected values, and fail for anything unexpected (default switch case)

Reply:
Actually I would just fix it in these.

Reply:
should I write a different test for PlatformNotSupportedException ?. Currently I am checking for default exitcode in case of this exception.